### PR TITLE
Adding swagger for Microsoft.OperationsManagement RP

### PR DESCRIPTION
**Pull request by dashimi16**
Also ran the Autorest tool to make sure that there is not Error.

### PR information
- [ x] The title of the PR is clear and informative.
- [ x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x ] My spec meets the review criteria:
  - [ x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/examples/OperationsList.json
Comment (jhendrixMSFT) [inline specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/examples/OperationsList.json]: There are JSON parse errors in the semantic and model validation, see the [travis](https://travis-ci.org/Azure/azure-rest-api-specs/builds/269677721?utm_source=github_status&utm_medium=notification) log.  While it's not clear from the error message I believe it's due to this trailing comma.

**Comment by jhendrixMSFT** [comment]
There are some linter warnings, see the linter [log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/269677733).  We wouldn't hold up this PR due to them however you might want to fix them now so you don't accrue clean-up debt.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
JSON parse errors must be fixed.

**Comment by jhendrixMSFT** [comment]
This change has introduced new errors, see the following.
```
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/270116255
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/270116256
```
I would recommend running the validation tools locally before committing changes, see [this](https://github.com/Azure/adx-documentation-pr/wiki/Azure-Swagger-Tools) link for instructions.

**Comment by jhendrixMSFT** [comment]
Getting closer.  There are some semantic validation errors that must be fixed, see log https://travis-ci.org/Azure/azure-rest-api-specs/jobs/270509581

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (jhendrixMSFT) [inline specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: I think this should be "Solution_CreateOrUpdate", @ravbhatnagar would you agree?

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (jhendrixMSFT) [inline specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: Should be Solution_Delete.

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (jhendrixMSFT) [inline specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: Should be Solution_Get.

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (jhendrixMSFT) [inline specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: Should be Solution_ListByResourceGroup.

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (dashimi16) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: If I make it singular it gives me warning since I have Solution in definition section. Should I use format like Solutions_Get, Solutions_Delete, etc.?

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (dashimi16) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: Solution_Get or Solutions_Get?

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (jhendrixMSFT) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: You are correct the noun here should be plural, see the [guidelines](https://github.com/Azure/azure-rest-api-specs/blob/current/documentation/openapi-authoring-automated-guidelines.md#r2063-operationidnounconflictingmodelnames).  Good catch and sorry for the confusion.

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (dashimi16) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: Solution_CreateOrUpdate or Solutions_CreateOrUpdate

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (jhendrixMSFT) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: Solutions_Get

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (jhendrixMSFT) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: Solutions_CreateOrUpdate

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (dashimi16) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: No Worries. Thanks.

**Comment by jhendrixMSFT** [comment]
This looks good, once we have ARM sign-off from @ravbhatnagar we can merge.

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (ravbhatnagar) [inline specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: tags property missing

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (ravbhatnagar) [inline specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: enum

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (ravbhatnagar) [inline specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: add a property called resourceId for the items array.

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (ravbhatnagar) [inline specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: add a property called resourceId for the items array.

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (ravbhatnagar) [inline specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: Collection GET at subscription scope missing.

Comment (ravbhatnagar) [inline specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: PATCH api missing

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (dashimi16) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: We did not implemented PATCH. This API is in preview and on a path to be deprecated. We do not want to spend any more time to change the API at this moment, The API is already checked in for a while.

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (dashimi16) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: We did not implemented it as an Enum.

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (dashimi16) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: We did not implemented tags for this and there is no plan to invest in this API any more since this is on it's path to be deprecated.

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (dashimi16) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: Why add this property? We have the Id property already.

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (dashimi16) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: We did not implemented tags

File: specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json
Comment (dashimi16) [reply specification/operationsmanagement/resource-manager/Microsoft.OperationsManagement/2015-11-01-preview/OperationsManagement.json]: Yes, I missed that. Will update the pull request with that.

**Comment by jhendrixMSFT** [comment]
@ravbhatnagar The missing GET has been added.  Based on @dashimi16 answers to your other questions do you approve the change?

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
@ravbhatnagar I'd like to merge this today however you're review is still marked as "changes requested".

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by dashimi16** [comment]
@ravbhatnagar, can you please let us know? Thanks.

**Review by ravbhatnagar — APPROVED** [review]

**Comment by ravbhatnagar** [comment]
Signing off!

**Pull request merged**
